### PR TITLE
Exercise: Historical life expectancy

### PR DESCRIPTION
**Pull request by artem-konovalov-personal**
Closed [#267](https://github.com/nesterone/sf-eng-init-gl-spring-2016/issues/267)

### CODE REVIEW

**Comment by artem-konovalov-personal** [comment]
I have changed code structure for this solution and add a function `lifeExpactancy`, but I had to closed the previous pull request and to create a new one due to some conflicts which were in my branch since least merge

File: 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/* global ancestry ancestryByCentury average lifeExpectancy */
+
+var ancestryByCentury = {};
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js]: It's not required to have that one outside of the function

File: 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js [hunk 2]
```diff
@@ -8,7 +12,30 @@ function average(array) {
 console.log(average([2, 2, 2]));
 // → 2
 
-// → Your code here.
+function lifeExpectancy(ancestry) {
+  var century;
+  var arr = [];
+  var key;
+
+  ancestry.forEach(function (property) {
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js]: `reduce` ?

File: 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js [hunk 3]
```diff
@@ -8,7 +12,30 @@ function average(array) {
...
+    century = Math.ceil(property.died / 100);
+    if (!ancestryByCentury[century]) {
+      ancestryByCentury[century] = [];
+    }
+
+    ancestryByCentury[century].push(property.died - property.born);
+  });
+
+  for (key in ancestryByCentury) {
```
Comment (nesterone) [inline 01-syntax/04-high-order-functions/life-expectancy/lifeExpectancy.js]: Could we go without 'for' loop here ?

**Comment by artem-konovalov-personal** [comment]
I have changed code structure: removed the `for` loop and add the `reduce` method

**Pull request merged**